docs/runtime/compiler-ir.md

```diff
@@ -9,6 +9,24 @@ multiple passes. Every pass is a class implementing the
 See [Runtime roadmap - static analysis](../runtime-roadmap.md#static-analysis)
 for future goals.
 
+## Linting
+
+If `org.enso.compiler.data.CompilerConfig#isLintingDisabled` is false, various
+additional _linting_ compiler passes are enabled in `org.enso.compiler.Passes`.
+Such passes usually generate warnings, for example, about an unused local
+binding. In production (when running the engine from the `language-server`),
+linting is disabled.
+
+### Removing unused imports
+
+Automatic removal of unused imports can be enabled with
+`enso.compiler.RemoveUnusedImports` system property. To remove unused imports
+from the whole `Standard.Base`, run:
```

Comment:
- it is great to have such a _self healing_ functionality
- better than just printing a warning to be applied manually
- however, after this initial batch of fixes is applied 
   - we won't really need the automatic rewrite again
   - CI will ensure we are clean before integration
   - violations reported by CI will be easy to fix manually
- e.g. such _write fixes down_ support could be left out of this PR
- and included properly in the future
- for example when #13080 is to be addressed

Reply:
Will be addressed in the follow-up PR.